tests/unit/test_optimum_utils.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2021 The HuggingFace Team, Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import tempfile
+
+import pytest
+from transformers.testing_utils import require_torch
+
+from sagemaker_huggingface_inference_toolkit.optimum_utils import (
+    get_input_shapes,
+    get_optimum_neuron_pipeline,
+    is_optimum_neuron_available,
+)
+from sagemaker_huggingface_inference_toolkit.transformers_utils import _load_model_from_hub
+
+
+require_inferentia = pytest.mark.skipif(
+    not is_optimum_neuron_available(),
+    reason="Skipping tests, since optimum neuron is not available or not running on inf2 instances.",
+)
+
+
+REMOTE_NOT_CONVERTED_MODEL = "hf-internal-testing/tiny-random-BertModel"
+REMOTE_CONVERTED_MODEL = "optimum/tiny_random_bert_neuron"
+TASK = "text-classification"
+
+
+@require_torch
+@require_inferentia
+def test_not_supported_task():
+    os.environ["HF_TASK"] = "not-supported-task"
+    with pytest.raises(Exception):
+        get_optimum_neuron_pipeline(task=TASK, model_dir=os.getcwd())
+
+
+@require_torch
+@require_inferentia
+def test_get_input_shapes_from_file():
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        storage_folder = _load_model_from_hub(
+            model_id=REMOTE_CONVERTED_MODEL,
+            model_dir=tmpdirname,
+        )
+        input_shapes = get_input_shapes(model_dir=storage_folder)
+        assert input_shapes["batch_size"] == 1
+        assert input_shapes["sequence_length"] == 16
+
+
+@require_torch
+@require_inferentia
```

Comment:
I don't think some of these tests actually need `@require_inferentia` to run successfully.

Are `@require_inferentia` tests going to be run in CI, in practice? If not, when will these actually be run?

Reply:
They are not going to run if neuron is not installed due to `is_optimum_neuron_available` for now we have to run the manually in this repository since there is no CI with inf2 yet. 
The DLC repository will have a test as well where it runs in the CI.